### ISSUE TITLE
Labels position

### ISSUE THREAD

**Issue by RobertoDF** [brainglobe/brainrender#176]
Hi Federico,
Labels are misplaced and upside down. You can reproduce it using your example add_labels.py.

```
from brainrender import Scene


from rich import print
from myterial import orange
from pathlib import Path


print(f"[{orange}]Running example: {Path(__file__).name}")


# crate a scene and add brain regions
scene = Scene()
th, mos = scene.add_brain_region("TH", "MOs")
scene.add_label(th, "TH")
scene.add_label(mos, "My region")

# render
scene.render()
```
![Screenshot 2021-11-11 at 17 12 45](https://user-images.githubusercontent.com/37729096/141331147-a740a570-3fcb-4362-a536-43e4b3ec044d.png)

The upside down is solved easily by changing:
https://github.com/brainglobe/brainrender/blob/19c63b97a34336898871d66fb24484e8a55d4fa7/brainrender/actor.py#L73 to 
`txt.pos(x=0, y=0, z=0).rotateX(0).rotateY(180).pos(p) ` 
but I couldn't solve the position yet, do you have any ideas?

I'm using brainrender 2.0.4.4 on Ubuntu.

Cheers!

**Comment by RobertoDF** [comment]
A z(third value) offset of 10000 puts the labels in the right position but the dots remains misplaced, I think something is wrong with the closestPoint method here: https://github.com/brainglobe/brainrender/blob/19c63b97a34336898871d66fb24484e8a55d4fa7/brainrender/actor.py#L79-L80

**Comment by RobertoDF** [comment]
It's much closer to the target area by making the third number of the "point" variable negative:
https://github.com/brainglobe/brainrender/blob/19c63b97a34336898871d66fb24484e8a55d4fa7/brainrender/actor.py#L52
by adding this afterwards:
`point[2] = -point[2]`

However the label is still off if I want to mark a probe represented just by points......

**Comment by FedeClaudi** [comment]
Oh dear, that's not good. Thanks for reporting it. 

I'll need to look into it but I'm currently on holiday, is this urgent? Is the workaround you've posted above good enough for a temporary fix?

**Comment by RobertoDF** [comment]
No worry, it's not urgent. Enjoy your holidays!

**Comment by RobertoDF** [comment]
After some more playing around I realised that by  changing this: https://github.com/brainglobe/brainrender/blob/19c63b97a34336898871d66fb24484e8a55d4fa7/brainrender/actor.py#L79-L81 to:
```
            pt = actor.closestPoint(point)
            pt[2] = -pt[2]
            sphere = Sphere(pt, r=radius, c=color)
            sphere.ancor = pt
```

I get the label's dots in the right place (the labels are built based on the part of the probe outside the brain)
![Screenshot 2021-11-13 at 12 56 02](https://user-images.githubusercontent.com/37729096/141642846-a4a4f13e-2a04-4ac7-a35e-24aa6e31c317.png) 
However, I cannot change the color of the dots! I think this is not a problem of brainrender though.
Btw I'm on vedo 2021.0.7 and vtk 8.1.2.

**Comment by FedeClaudi** [comment]
Awesome! If you'd like you can open a pull request with this bug fix, otherwise I can implement it in the near feature.
Thank you very much for working on finding a solution.

**Comment by FedeClaudi** [comment]
Hi Roberto,

I've managed to fix the labels!

<img width="1364" alt="image" src="https://user-images.githubusercontent.com/17436313/142629480-e8189401-e2f3-4194-841a-832dc77ff2f0.png">


Your fix was very helpful, after that I've removed the `followCamera()` bit from `Text2D`. That's supposed to keep the text facing the camera regardless of the camera orientation, but I thought it didn't look to good for the brainrender scene.

Fixing the label dot color was more complicated. Brainrender applies a matrix transform to each mesh before rendering, this results in the meshes being reversed inside/out. That's a problem for 3D rendering, so normally I reverse the orientation of each mesh before rendering. However, this was not done for the dot that made up the labels, which meant that we were always seeing the backface and not the outside, colored face. Once I've added the reverse to the dot as well everything worked fine.

Thank you again for the help!
